pages/api/id/[form]/submission/report.ts

```diff
@@ -0,0 +1,245 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, jsonValidator, sessionExists } from "@lib/middleware";
+import submissionNamesArraySchema from "@lib/middleware/schemas/submission-name-array.schema.json";
+import {
+  BatchGetCommand,
+  DynamoDBDocumentClient,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps, WithRequired } from "@lib/types";
+import { connectToDynamo } from "@lib/integration/dynamodbConnector";
+import { NotifyClient } from "notifications-node-client";
+import { checkOne } from "@lib/cache/flags";
+
+class FailedToSendEmailThroughGCNotify extends Error {}
+
+const handler = async (req: NextApiRequest, res: NextApiResponse, props: MiddlewareProps) => {
+  // Is this feature / endpoint active
+  const vaultActive = await checkOne("vault");
+  if (!vaultActive) return res.status(404).json({ error: "Vault not active" });
+
+  const { session } = props as WithRequired<MiddlewareProps, "session">;
+
+  const userEmail = session.user.email;
+  if (userEmail === null)
+    throw new Error(
+      `User does not have an associated email address: ${JSON.stringify(session.user)} `
+    );
+
+  const formId = req.query.form;
+
+  if (Array.isArray(formId) || !formId) return res.status(400).json({ error: "Bad request" });
+
+  const submissionNames = req.body as string[];
```

Comment:
Same as with the confirm API we should add a limit to the number of codes that can be passed in at once.